metadatastore/conf.py

```diff
@@ -3,18 +3,29 @@
 import logging
 
 logger = logging.getLogger(__name__)
-filename = os.path.join(os.path.expanduser('~'), '.config', 'metadatastore',
-                        'connection.yml')
-if os.path.isfile(filename):
-    with open(filename) as f:
-        connection_config = yaml.load(f)
-    logger.debug("Using db connection specified in config file. \n%r",
-                 connection_config)
-else:
-    connection_config = {
-        'database': 'test',
-        'host': "localhost",
-        'port': 27017,
-        }
-    logger.debug("Using default db connection. \n%r",
-                 connection_config)
+
+def load_configuration(name, prefix, fields):
+    filenames = [os.path.join('etc', name + '.yml'),
+                 os.path.join(os.path.expanduser('~'), '.config',
```

Comment:
+1 on splitting the package defaults from the config file exposed to users.

It's nice when an installer:
      - checks to see if a config file already exists
      - installs a well commented config file, with an additional suffix
        if the file already exists (ala. ".rpmnew" for rpm packages)
      - ideally no default passwords for people to forget to change;
        setup with an init script instead